src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1529,6 +1530,30 @@ public function getContent($asResource = false)
         return $this->content;
     }
 
+    /**
+     * Return request body content as a json decoded array.
+     *
+     * @return array
+     *
+     * @throws JsonException|\LogicException
+     */
+    public function getJson()
+    {
+        $body = $this->getContent();
+
+        if (!$body) {
+            throw new \LogicException('Empty request body cannot be decoded to JSON');
```

Comment:
could be `JsonException` as well :thinking:

Reply:
Should be!

Reply:
For me, there is a difference between not being able to decode JSON data, and trying to decode something that doesn't exist (E.G calling `getJson` during `GET` request). When this exception is triggered, it means the calling code is probably doing something strange or missing some checks

Reply:
but we dont do any of those checks here; i can POST an empty string as well.

Given getContent() is not nullable, IMHO there's no difference; if we cant parse the content as JSON, it's a JsonException.

Reply:
or put different, i think treating empty content as a  logic error is a responsibility of the calling side. Right now you need to know about to cases to catch _by default_, which can even be a trap for the user.

Reply:
In that case, this check can then probably be dropped altogether and let the error handling below handle the empty string